### DISCUSSION TITLE
Easiest way to log java.time.Instant (and similar) as strings

### DISCUSSION THREAD

**Discussion by djarnis73** [logfellow/logstash-logback-encoder#1044] [Q&A] [answered]
Hi

I have been googling around looking for at simple way to configure the encoder to log structured objects containing `java.time.Instant` fields as strings.

For instance if I log like:

```
log.atInfo().addKeyValue("event", event).log("Event handled");
```
any `java.time.Instant` fields in the `event` object are logged as decimal value like `1724834916.765`, and I would prefer an easier human readable form, like for instance `"2024-08-28T12:41:30.459867Z"`.

Can this be done without a custom Decorator?

In a project that uses the ObjectMapper this can be achieved by configuring like: 
```
ObjectMapper objectMapper = new ObjectMapper()
            .registerModule(new JavaTimeModule())
            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
```
I did search google / issues / PRs / discussions without finding anything.

**Comment by djarnis73** [comment]
Just to quickly answer my own question. The example in the readme (using a project local `ISO8601DateDecorator`) works fine. I was just wondering if there was a way to do this without additional classes.

**Comment by djarnis73** [comment]
Would it make sense to add a `net.logstash.logback.decorate.FeatureObjectMapperDecorator` where you can enable/disable features on the `ObjectMapper`? I think I would be able to contribute that if it would be accepted.

**Reply by philsttr** [reply, marked as answer]
logstash-logback-encoder 9.0 has a `DateTimeFeatureDecorator` you can use to enable/disable `DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS`
